frontend/components/App.tsx

```diff
@@ -83,75 +62,37 @@ const AppContainer = styled.div`
   padding: 0;
 `;
 
-// prettier-ignore lint//
 const Main = styled.main<{ $fullwidth: boolean }>`
   grid-area: ${({ $fullwidth }) => ($fullwidth ? '1 / 1 / -1 / -1' : 'main')};
   background: ${bgColor};
   height: calc(100vh - (2 * ${defaultMargin}));
-  max-width: ${({ $fullwidth }) =>
-    $fullwidth ? '' : `calc(90vw - ${sidebarWidth} )`};
+  max-width: ${({ $fullwidth }) => $fullwidth ? '' : `calc(90vw - ${sidebarWidth})`};
   padding: ${defaultMargin} ${sidebarShowButtonSize};
   margin: 0;
 `;
 
+// Define the main App component
 function App() {
-  /**
-   * Reducers
-   * useMemo prevents rerenders when state does not change. necessary because of useContext
-   */
-  const [menuState, menuDispatch] = useReducer(menuReducer, initialMenuState);
-  const menuProvider = useMemo(
-    () => ({ state: menuState, dispatch: menuDispatch }),
-    [menuState],
-  );
 
-  // initializing the initial viewState object
-  // this is the app views that will be passed through a provider to any children components wrapped in it. Right now, only sidebar is wrapped in it.
-  const initialAppViewState: AppViewState = {
-    selectedView: 'dbView',
-    sideBarIsHidden: false,
-    showConfigDialog: false,
-    showCreateDialog: false,
-    PG_isConnected: false,
-    MYSQL_isConnected: false,
-  };
+  const dispatch = useDispatch(); 
```

Comment:
Please add some comments to the dispatch statements to describe how and where it is going to be used. Please also export those statements if they are going to be used in other files